lib/conduit_async_ssl.mli

```diff
@@ -33,7 +33,7 @@ val ssl_connect :
   ?verify:(Ssl.Connection.t -> bool Deferred.t) ->
   Reader.t ->
   Writer.t ->
-  (Reader.t * Writer.t) Deferred.t
+  (Reader.t * Writer.t) Deferred.Or_error.t
```

Comment:
On 01/06/2016 19:20, Rudi Grinberg wrote:
> Are you sure about |Pipe.create|? I'm pretty sure pipes aren't tied to
> any particular fd's. It's readers/writers that are fd wrappers.

Yeah you're right, they're not tied to fds, my bad. So this is even
safer that what I said :)

> But anyway if you're reasonably certain then I support this change.

Yeah, reasonably certain… :) But in the end I agree that it's tricky to
be sure to not breach this contract, I just wanted to do something
similar as Ssl.client that returns an Or_error.t.

The |verify| function could raise, for example. And indeed the
|Reader.of_pipe| can raise if there is no fd left.

I'm not certain anymore what's the best thing to do. I think best would
be wrap everything in a big try_with or do not return Or_error.t types.
What do you think?